src/microcosm.js

```diff
@@ -121,13 +138,30 @@ export default class Microcosm extends Emitter {
   rollforward () {
     this.history.prune(this.clean, this)
 
-    this.state = this.history.reduce(this.dispatch, this.cache, this)
+    const staged = this.history.reduce(this.dispatch, this.archive, this)
+
+    const next = this.stores.reduce((memo, store) => {
+      return this.commit(memo, store[0], store[1])
+    }, merge({}, staged))
```

Comment:
I don't like this. This is inefficiency (though not much). I don't like shallow copying the staged changes, and I don't like the anonymous function call. I also don't like the `[key, value]` tuple. 

Something to muse on...